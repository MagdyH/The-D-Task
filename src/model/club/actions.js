import ActionType from './constants';

export const Actions = {
    getAllClubs: (clubs)=>{
        return {type:ActionType.GET_ALL_CLUB,clubs}
    },
    searchClubs:(searchTirm)=>{
        return {type:ActionType.SEARCH_CLUB,searchTirm}
    },
    applySearchMode:(searchMode)=>{
        return {type:ActionType.SEARCH_MODE,searchMode}
    }
}
export default Actions;
