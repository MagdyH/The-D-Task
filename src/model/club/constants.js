export const ActionType = {
    GET_ALL_CLUB: 'club/GET_ALL_CLUB',
    SEARCH_CLUB: 'club/SEARCH_CLUB',
    SEARCH_MODE : 'common/SEARCH_MODE'
}

export default ActionType;
