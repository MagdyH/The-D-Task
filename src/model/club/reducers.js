import ActionType from './constants';

export default function clubReducer(state = { clubs: [], storedClubs: [],searchMode:'' }, action) {
    switch (action.type) {
        case ActionType.GET_ALL_CLUB:
            return { ...state, clubs: action.clubs, storedClubs: action.clubs }
        case ActionType.SEARCH_CLUB:
            let result = [];
            if (action.searchTirm === "" || action.searchTirm === undefined) {
                result = state.storedClubs;
            }
            else {
                if (state.storedClubs.length > 0)
                    result = state.storedClubs.filter((club) => club.team_name.toLowerCase().includes(action.searchTirm.toLowerCase()));
            }
            return { ...state, clubs: result }
        case ActionType.SEARCH_MODE:
            return { ...state, searchMode: action.searchMode }
        default:
            return state;
    }
}