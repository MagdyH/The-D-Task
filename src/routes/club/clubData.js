import * as React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PlayerCard from '../../components/playerCard';
import API from '../../api/footballAPI';
import Actions from '../../model/club/actions';
import GameCard from '../../components/gameCard';
import Utilities from '../../utilities/utility';

/*
container component to show club plyers and next,previous games.
*/
export function ClubData(props) {
    let latestGames = [];
    let nextGames = [];
    let list = [];

    var date = new Date();
    
    var club = props.clubs ? props.clubs.filter((c) => c.team_key === props.match.params.clubId) :[];

    if (club.length > 0) {

        club[0].coaches.map((manager) => list.push(<PlayerCard key={manager.coach_name} manager={manager} />))
        club[0].players.map((player) => list.push(<PlayerCard key={player.player_key} player={player} />))
        nextGames = props.games.filter((g) => Utilities.convertDate(g.match_date) >= Utilities.convertDate(date) && (g.match_hometeam_id === club[0].team_key || g.match_awayteam_id === club[0].team_key));
        latestGames = props.games.filter((g) => Utilities.convertDate(g.match_date) < Utilities.convertDate(date) && (g.match_hometeam_id === club[0].team_key || g.match_awayteam_id === club[0].team_key));
    }

    return (
        <div className="container club-data">
            <div className="row">
                <div className="col-md-9">
                    {club.length > 0 &&
                        <div className="card my-4 mx-2 col-md-6">
                            <div className='row card-body'>
                                <div className="p-0 w-25">
                                    <img alt={''} src={club[0].team_badge} className="img-responsive player-img" />
                                </div>
                                <div className="col-sm-9">
                                    <h1 className="card-title">{club[0].team_name}</h1>
                                    <p className="card-text">{club[0].team_name}</p>
                                </div>

                            </div>
                        </div>
                    }
                    {nextGames.length > 0 &&
                        <div>
                            <h1>Next Game</h1>
                            <div col-md-6>
                                {nextGames.map((g) => <GameCard game={g} />)}
                            </div>
                        </div>
                    }
                    {latestGames.length > 0 &&
                        <div>
                            <h1>Latest Game</h1>
                            <div col-md-6>
                                {latestGames.map((g) => <GameCard game={g} />)}
                            </div>
                        </div>
                    }
                </div>

                <div className="my-4 col-md-3 Scrollable ">
                    {list}
                </div>
            </div>
        </div>
    )
}

function mapStateToProps(state, ownProps) {
    return {
        clubs: state.club ? state.club.clubs:[],
        games: state.game ? state.game.games:[]
    }
}


function mapDispatcherToProps(dispatch) {
    return {
        API: bindActionCreators(API, dispatch),
        Actions: bindActionCreators(Actions, dispatch)
    }

}

export default connect(mapStateToProps, mapDispatcherToProps)(ClubData);