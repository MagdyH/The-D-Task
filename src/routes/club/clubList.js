import * as React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import API from '../../api/footballAPI';
import Actions from '../../model/club/actions';
import ClubCard from '../../components/clubCard'
/*
container component to show clubs list.
*/
export function ClubList(props) {
    if(props.Actions){
        props.Actions.applySearchMode('Clubs');
    }
    return (
        <div className='container'>
            <div className='row'>
                <div className='row col-md-12'>
                {(props.clubs && props.clubs.length) > 0 &&
                    props.clubs.map((club) => {
                        return <div className='col-md-3'>
                            <ClubCard key={club.team_key} club={club} />
                        </div>
                    
                    })
                }
                </div>
            </div>
        </div>
    )
}

function mapStateToProps(state, ownProps) {
    return {
        clubs: state.club? state.club.clubs:[]
    }
}


function mapDispatcherToProps(dispatch) {
    return {
        API: bindActionCreators(API, dispatch),
        Actions: bindActionCreators(Actions, dispatch)
    }

}

export default connect(mapStateToProps, mapDispatcherToProps)(ClubList);