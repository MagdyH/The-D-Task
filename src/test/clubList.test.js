import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

import {ClubList as Clubs} from '../routes/club/clubList';


describe('ClubList component', () => {
    it('should render snapshot', () => {
      const component = renderer.create(<Clubs />);
  
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

});