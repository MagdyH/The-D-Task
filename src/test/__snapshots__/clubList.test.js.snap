// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClubList component should render snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="row col-md-12"
    />
  </div>
</div>
`;
